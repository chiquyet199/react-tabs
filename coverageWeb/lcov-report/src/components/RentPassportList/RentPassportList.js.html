<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/RentPassportList/RentPassportList.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/RentPassportList</a> RentPassportList.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import {View} from "react-native";
import {
  Container,
  RentPassportListItemContainer,
  RentPassportListItemIcon,
  RentPassportListItemDetails,
  Status,
  StatusIcon,
  StatusText,
  ListItemSpacer,
  Title,
  Info,
} from "./RentPassportList.styles";
import {bool, number, oneOfType, shape, string} from "prop-types";
&nbsp;
export default Object.assign(
  ({complete = <span class="branch-0 cbranch-no" title="branch not covered" >false,</span> item}) =&gt; (
    &lt;Container&gt;
      &lt;RentPassportListItemContainer style={{marginTop: 10}}&gt;
        &lt;ListItemSpacer&gt;
          &lt;RentPassportListItemIcon
            complete={complete}
            source={item.icons.main}
          /&gt;
        &lt;/ListItemSpacer&gt;
        &lt;RentPassportListItemDetails&gt;
          &lt;View&gt;
            &lt;Title complete={complete}&gt;{item.title}&lt;/Title&gt;
            &lt;Info complete={complete}&gt;{item.infoText}&lt;/Info&gt;
          &lt;/View&gt;
        &lt;/RentPassportListItemDetails&gt;
      &lt;/RentPassportListItemContainer&gt;
      &lt;RentPassportListItemContainer&gt;
        &lt;ListItemSpacer /&gt;
        &lt;Status&gt;
          &lt;StatusIcon
            complete={complete}
            name={complete ? item.icons.complete : item.icons.incomplete}
          /&gt;
          &lt;StatusText complete={complete}&gt;
            {complete ? item.statusText.complete : item.statusText.incomplete}
          &lt;/StatusText&gt;
        &lt;/Status&gt;
      &lt;/RentPassportListItemContainer&gt;
    &lt;/Container&gt;
  ),
  {
    propTypes: {
      complete: bool.isRequired,
      item: shape({
        title: string.isRequired,
        infoText: string.isRequired,
        icons: shape({
          main: oneOfType([number.isRequired, string.isRequired]),
          complete: string.isRequired,
          incomplete: string.isRequired,
          collapsed: oneOfType([number.isRequired, string.isRequired]),
          open: oneOfType([number.isRequired, string.isRequired]),
        }),
        statusText: shape({
          complete: string.isRequired,
          incomplete: string.isRequired,
        }),
      }),
    },
  },
);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:06 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
