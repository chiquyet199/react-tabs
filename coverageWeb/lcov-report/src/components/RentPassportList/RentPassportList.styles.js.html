<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/RentPassportList/RentPassportList.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/RentPassportList</a> RentPassportList.styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.11% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.11% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/18</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from "styled-components/native";
import Icon from "../Icon";
&nbsp;
export const Container = styled.View`
  flex: 1;
  border-bottom-width: 1px;
  border-bottom-color: ${<span class="fstat-no" title="function not covered" >({</span>theme}) =&gt; <span class="cstat-no" title="statement not covered" >theme.colors.lightSteel}</span>;
  margin-horizontal: 8;
`;
&nbsp;
export const RentPassportListItemContainer = styled.View`
  flex: 4;
  flex-direction: row;
`;
&nbsp;
export const RentPassportListItemIcon = styled.Image`
  height: 64px;
  width: 64px;
  opacity: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; (<span class="cstat-no" title="statement not covered" >props.complete ? 1 : 0.2)</span>};
`;
&nbsp;
export const RentPassportListItemDetails = styled.View`
  flex: 10;
  justify-content: space-evenly;
`;
&nbsp;
export const Status = styled.View`
  flex: 10;
  flex-direction: row;
  align-items: center;
`;
&nbsp;
export const StatusIcon = styled(Icon)`
  margin-right: 5px;
  padding-top: 2px;
  color: ${<span class="fstat-no" title="function not covered" >({</span>complete, theme}) =&gt;
<span class="cstat-no" title="statement not covered" >    complete ? theme.colors.canopyGreen : theme.colors.lightSteel}</span>;
`;
&nbsp;
export const StatusText = styled.Text`
  font-size: 12px;
  color: ${<span class="fstat-no" title="function not covered" >({</span>complete, theme}) =&gt;
<span class="cstat-no" title="statement not covered" >    complete ? theme.colors.canopyGreen : theme.colors.lightSteel}</span>;
`;
&nbsp;
export const ListSpace = styled.View`
  flex: 5;
  width: 100%;
  background-color: ${<span class="fstat-no" title="function not covered" >({</span>theme}) =&gt; <span class="cstat-no" title="statement not covered" >theme.colors.white}</span>;
`;
&nbsp;
export const ListItemSpacer = styled.View`
  flex: 2;
  justifycontent: center;
  alignitems: center;
`;
&nbsp;
export const Title = styled.Text`
  font-size: 24px;
  color: ${<span class="fstat-no" title="function not covered" >({</span>complete, theme}) =&gt;
<span class="cstat-no" title="statement not covered" >    complete ? theme.colors.canopyGreen : theme.colors.lightSteel}</span>;
`;
&nbsp;
export const Info = styled.Text`
  font-size: 14px;
  color: ${<span class="fstat-no" title="function not covered" >({</span>complete, theme}) =&gt;
<span class="cstat-no" title="statement not covered" >    complete ? theme.colors.canopyGreen : theme.colors.lightSteel}</span>;
`;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:06 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
