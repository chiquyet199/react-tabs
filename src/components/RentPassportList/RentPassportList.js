import React from "react";
import {View} from "react-native";
import {
  Container,
  RentPassportListItemContainer,
  RentPassportListItemIcon,
  RentPassportListItemDetails,
  Status,
  StatusIcon,
  StatusText,
  ListItemSpacer,
  Title,
  Info,
} from "./RentPassportList.styles";
import {bool, number, oneOfType, shape, string} from "prop-types";

export default Object.assign(
  ({complete = false, item}) => (
    <Container>
      <RentPassportListItemContainer style={{marginTop: 10}}>
        <ListItemSpacer>
          <RentPassportListItemIcon
            complete={complete}
            source={item.icons.main}
          />
        </ListItemSpacer>
        <RentPassportListItemDetails>
          <View>
            <Title complete={complete}>{item.title}</Title>
            <Info complete={complete}>{item.infoText}</Info>
          </View>
        </RentPassportListItemDetails>
      </RentPassportListItemContainer>
      <RentPassportListItemContainer>
        <ListItemSpacer />
        <Status>
          <StatusIcon
            complete={complete}
            name={complete ? item.icons.complete : item.icons.incomplete}
          />
          <StatusText complete={complete}>
            {complete ? item.statusText.complete : item.statusText.incomplete}
          </StatusText>
        </Status>
      </RentPassportListItemContainer>
    </Container>
  ),
  {
    propTypes: {
      complete: bool.isRequired,
      item: shape({
        title: string.isRequired,
        infoText: string.isRequired,
        icons: shape({
          main: oneOfType([number.isRequired, string.isRequired]),
          complete: string.isRequired,
          incomplete: string.isRequired,
          collapsed: oneOfType([number.isRequired, string.isRequired]),
          open: oneOfType([number.isRequired, string.isRequired]),
        }),
        statusText: shape({
          complete: string.isRequired,
          incomplete: string.isRequired,
        }),
      }),
    },
  },
);
