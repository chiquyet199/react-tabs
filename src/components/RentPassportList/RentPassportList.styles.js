import styled from "styled-components/native";
import Icon from "../Icon";

export const Container = styled.View`
  flex: 1;
  border-bottom-width: 1px;
  border-bottom-color: ${({theme}) => theme.colors.lightSteel};
  margin-horizontal: 8;
`;

export const RentPassportListItemContainer = styled.View`
  flex: 4;
  flex-direction: row;
`;

export const RentPassportListItemIcon = styled.Image`
  height: 64px;
  width: 64px;
  opacity: ${props => (props.complete ? 1 : 0.2)};
`;

export const RentPassportListItemDetails = styled.View`
  flex: 10;
  justify-content: space-evenly;
`;

export const Status = styled.View`
  flex: 10;
  flex-direction: row;
  align-items: center;
`;

export const StatusIcon = styled(Icon)`
  margin-right: 5px;
  padding-top: 2px;
  color: ${({complete, theme}) =>
    complete ? theme.colors.canopyGreen : theme.colors.lightSteel};
`;

export const StatusText = styled.Text`
  font-size: 12px;
  color: ${({complete, theme}) =>
    complete ? theme.colors.canopyGreen : theme.colors.lightSteel};
`;

export const ListSpace = styled.View`
  flex: 5;
  width: 100%;
  background-color: ${({theme}) => theme.colors.white};
`;

export const ListItemSpacer = styled.View`
  flex: 2;
  justifycontent: center;
  alignitems: center;
`;

export const Title = styled.Text`
  font-size: 24px;
  color: ${({complete, theme}) =>
    complete ? theme.colors.canopyGreen : theme.colors.lightSteel};
`;

export const Info = styled.Text`
  font-size: 14px;
  color: ${({complete, theme}) =>
    complete ? theme.colors.canopyGreen : theme.colors.lightSteel};
`;
