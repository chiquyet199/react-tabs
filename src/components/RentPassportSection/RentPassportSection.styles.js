import styled from "styled-components/native/index";

export const Title = styled.Text`
  font-size: 24px;
  color: ${({theme}) => theme.textColor};
`;
