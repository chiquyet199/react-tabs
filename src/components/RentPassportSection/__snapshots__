/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/RentPassportSection/__native_snapshots__