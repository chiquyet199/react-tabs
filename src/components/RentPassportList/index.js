import React from "react";
import {ListSpace} from "./RentPassportList.styles";
import RentPassportItem from "./RentPassportList";

const RentPassportList = ({items, complete}) => (
  <ListSpace>
    {items.map(item => (
      <RentPassportItem key={item.title} item={item} complete={complete} />
    ))}
  </ListSpace>
);

RentPassportList.defaultProps = {
  items: [],
};

export default RentPassportList;
