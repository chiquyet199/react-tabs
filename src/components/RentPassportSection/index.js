import React from "react";
import RentPassportItem from "../RentPassportsPreview/RentPassportsPreview";
import {View} from "react-native";
import {Title} from "./RentPassportSection.styles";
import {
  object,
  bool,
  func,
  string,
  arrayOf,
  shape,
  number,
  oneOf,
} from "prop-types";

const RentPassportSection = ({
  title,
  data,
  theme,
  selectedPassport,
  addPassportsToShare,
  globaltheme,
}) => (
  <View id="rent-passport-section">
    <Title>{title}</Title>
    {data.map(item => (
      <RentPassportItem
        key={item.id}
        theme={theme}
        globaltheme={globaltheme}
        item={item}
        selectedPassport={selectedPassport}
        addPassportsToShare={addPassportsToShare}
      />
    ))}
  </View>
);

export default RentPassportSection;

RentPassportSection.propTypes = {
  title: string.isRequired,
  data: arrayOf(
    shape({
      id: number.isRequired,
      name: string,
      pending: bool,
    }),
  ),
  theme: object.isRequired,
  selectedPassport: oneOf(["shared", "pending"]).isRequired,
  addPassportsToShare: func.isRequired,
};
