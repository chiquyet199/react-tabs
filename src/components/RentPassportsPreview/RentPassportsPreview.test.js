import React from "react";
import {shallow} from "enzyme";
import RentPassportList from "./";
import RentPassportSection from "../RentPassportSection";

describe("RentPassportList", () => {
  describe("List", () => {
    it("matches snapshot", () => {
      const wrapper = shallow(
        <RentPassportList data={sharedItem} {...mockProps()} />,
      );
      expect(wrapper).toMatchSnapshot();
    });

    it("renders RentPasportSection", () => {
      const wrapper = shallow(
        <RentPassportList data={sharedItem} {...mockProps()} />,
      );
      const section = wrapper.find(RentPassportSection);
      expect(section).toExist();
    });
  });
});

const sharedItem = [
  {
    id: 777,
    name: "Jonathan Beaumont",
    status: "Fully referenced",
  },
];

const mockProps = () => {
  return {
    title: "test",
    addPassportsToShare: () => {},
    selectedPassport: "shared",
    id: "RentPassportList",
    translations: {
      shared: "test",
      pending: "test",
    },
    theme: {textColor: "#fff"},
  };
};
