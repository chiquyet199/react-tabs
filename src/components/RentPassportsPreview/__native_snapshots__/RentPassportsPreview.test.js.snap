// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentPassportList List matches snapshot 1`] = `
<Styled(Component)
  theme={
    Object {
      "textColor": "#fff",
    }
  }
>
  <RentPassportSection
    addPassportsToShare={[Function]}
    data={
      Array [
        Object {
          "id": 777,
          "name": "Jonathan Beaumont",
          "status": "Fully referenced",
        },
      ]
    }
    selectedPassport="shared"
    theme={
      Object {
        "textColor": "#fff",
      }
    }
    title="test"
  />
</Styled(Component)>
`;
